Fix the issue of the missing (i) icon next to qtype in search widgets (issue #827)

juzen2003 opened the pull request:
- Fixes #827 
- Local branch: tooltip_issue_for_any_all_only_827
- Use the same popover method for the hover over event on (i) icon next to qtype.

juzen2003 committed:
Fix the missing (i) icon next to qtype in search widgets.
- In widget.js (line 425-434), initialize popover of (i) icon next to
  qtype. This is the required step to make popover work.
- In widget.html (line 47-51), add the (i) icon at the end of django form
  if it has a qtype option.
- In views.py (line 308), pass initial_qtype to widget.html, this is used
  to determine if we should render the (i) at the end of django form in
  html.
- In opus.css (line 1312, 1317), base.html (line 456), rename "range-qtype-helper"
  to "op-range-qtype-helper" and "range-qtype-helper-html" to "op-range-qtype-helper-html".
-In widget.html (line 4-5), update comments.

juzen2003 committed:
Remove unused commented out codes.
- In widgets.js (line 79), remove unwanted commented out codes.

juzen2003 committed:
Update the method to append help icon for any/all/only qtype.
- In widgets.js (line 412-422), append (i) icon if we have a widget with
  qtype options: any/all/only.
- In widget.html (line 46), remove the logic that append (i) icon for
  qtype, we will do it properly in widgets.js.
- In views.py (line 306), remove passing in initial_qtype to html.

juzen2003 committed:
Restore views.py to original file (same as master).
- In views.py (line 296), put back the extra line.

dstopp commented on opus/application/static_media/js/widgets.js:
```diff
@@ -409,6 +409,27 @@ var o_widgets = {
             }
 
+            // If we have a widget with qtype options any/all/only, we will append the (i) icon.
+            if ($(`#widget__${slug} select[name="${qtype}"]`).length !== 0) {
```
What do you think about combining the previous 'if' block and just do all the "${qtype}" business together?

dstopp commented on opus/application/apps/ui/templates/ui/base.html:
```diff
@@ -453,7 +453,7 @@ <h3 class='op-header-text'></h3>
 
     {% include "ui/footer.html" %}
-    <div id="range_query_widget_helper_html">
+    <div id="op-range-qtype-helper-html">
```
What do you think of renaming the id to "op-qtype-tooltip" instead?  I'm not clear that we need 'helper' or 'html' in that name and my thought is that this is just the tooltip for the qtype.

juzen2003 committed:
Update logic in callback function when getWidget is done, and rename
"op-range-qtype-helper-html" to "op-qtype-tooltip".
- In widgets.js (line 414-421), update logic in callback function when
  getWidget is done.
- In base.html (line 456), widget.html (line 4), opus.css (line 1317),
  widgets.js (line 430), rename "op-range-qtype-helper-html" to
  "op-qtype-tooltip".
